pkg/kapp/clusterapply/delete_change.go

```diff
@@ -110,3 +114,11 @@ func (c DeleteOrphanStrategy) Apply() error {
 	_, err = c.d.identifiedResources.Patch(c.res, types.JSONPatchType, patchJSON)
 	return err
 }
+
+func descMessage(res ctlres.Resource) []string {
+	if res.IsDeleting() {
```

Comment:
Added this check after observing this behaviour  - 

When deploying a Service resource (without finalizer) then Service and its Endpoint resource get deleted and exists come false for them but EndpointSlice do exist and deletionTimestamp is not present in existingRes of EndpointSlice so we have to add the condition to check for res.IsDeleting() for such usecases.